numpy/core/tests/test_multiarray.py

```diff
@@ -6457,7 +6457,7 @@ class TestArrayPriority(TestCase):
         op.ge, op.lt, op.le, op.ne, op.eq
         ]
 
-    if sys.version_info[0] < 3:
+    if sys.version_info[0] < 3 and not sys.py3kwarning:
```

Comment:
Could maybe use a comment. Or put the failing tests into a suppress_warnings context. I think I am fine with just doing it like this, but an explanation might help in the long run.